### PR TITLE
add memory widget

### PR DESCRIPTION
**Pull request by Mic92**

### CODE REVIEW

File: libqtile/widget/memory.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import re
```
Comment (flacjacket) [inline libqtile/widget/memory.py]: Please add a copyright notice at the top of this file (see the other files for an example)

File: libqtile/widget/memory.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+from . import base
+
+
+def get_meminfo():
+    val = {}
+    with open('/proc/meminfo') as file:
+        for line in file:
+            key, tail = line.split(':')
+            uv = tail.split()
+            val[key] = int(int(uv[0]) / 1000)
```
Comment (flacjacket) [inline libqtile/widget/memory.py]: You can do `int(uv[0]) // 1000` if you do a `from __future__ import division` at the top of the file

File: libqtile/widget/memory.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    val['MemUsed'] = val['MemTotal'] - val['MemFree']
+    return val
+
+
+class Memory(base.InLoopPollText):
+    """Displays a memory usage graph."""
```
Comment (flacjacket) [inline libqtile/widget/memory.py]: It looks like this displays text, not a graph.

File: libqtile/widget/memory.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    def __init__(self, **config):
+        super(Memory, self).__init__(**config)
+        self.add_defaults(Memory.defaults)
+
+    def poll(self):
+        mem = get_meminfo()
+
+        def replace(m):
+            return str(mem.get(m.group(1), "(nil)"))
+        return re.sub("%{([^}]+)}", replace, self.format)
```
Comment (flacjacket) [inline libqtile/widget/memory.py]: What exactly is this doing? Is this not something that you can do with `.format()` more easily? (cf. https://docs.python.org/3/library/string.html#format-examples)

**Comment by Mic92** [comment]
Add proposed changes. Squashed everything into one commit.

**Comment by flacjacket** [comment]
Great, thanks!

**Pull request closed without merging**